File: auth-service/src/main/kotlin/com/uitnetwork/authservice/template/Template.kt
package com.uitnetwork.authservice.template

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Template(
        @Id
        var id: String? = null,

        var name: String,

        var description: String,

        var html: Boolean,

        var hasParameter: Boolean,

        var templateParams: Set<TemplateParam> = emptySet(),

        var template: String,

        @JsonIgnore
        var userId: String = ""
)

data class TemplateParam(
        var paramKey: String,
        var required: Boolean
)

File: auth-service/src/main/frontend/src/app/models/template.ts
export class Template {
  constructor(public id: string, public name: string, public description: string, public html: boolean,
              public hasParameter: boolean, public template: string, public templateParams: Array<TemplateParam>) {
  }
}

export class TemplateParam {
  constructor(public paramKey: string, public required: boolean) {
  }
}
